exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,78 @@
+{ "exercise": "diffie-hellman"
+, "version" : "1.0.0"
+, "comments":
+    [ " In the following descriptions, p and g represent two          "
+    , " prime numbers.                                                "
+    , "                                                               "
+    , " Unless specified otherwise, tests are setup with              "
+    , " new DiffieHellman object using p = 23; g = 5.                 "
+    , "                                                               "
+    , " Alice's private key = 6                                       "
+    , " Alice's public key = 8                                        "
+    , "                                                               "
+    , " Bob's private key = 15                                        "
+    , " Bob's public key = 19                                         "
+    ]
+, "cases":
+    [ { "description": "Private keys are generated in Range (1..p)"
+      , "property"   : "generatePrivateKey"
+      , "p": 23
+      }
+    , { "description": "generatePrivateKey's output is random."
+      , "property"   : "generatePrivateKey"
+      }
+    , { "description": "Public key is correctly calculated from private key and two primes."
+      , "property"   : "getPublicKeyFromPrivateKey"
+      , "p": 23
+      , "g": 5
+      , "private_key": 6
+      , "expected": 8
+      }
+    , { "description": "Shared secret is correctly calculated."
+      , "property"   : "generateSharedSecret"
+      , "p": 23
+      , "bob_public_key": 19
+      , "alice_private_key": 6
+      , "expected": 2
+      }
+    , { "description": "Shared secret generated by Alice and Bob should be equal."
```

Comment:
I would have expected this to define `alice_public_key` and `bob_private_key` and `expected`, otherwise it is not clear what values we are testing.

If anyone reads my suggestion and is concerned that the same property `generateSharedSecret` may display two sets of input keys (`[alice_public_key, bob_private_key]` vs `[bob_public_key, alice_private_key]`) may I instead suggest just `public_key` and `private_key`?

If anyone reads that suggestion and now objects that it's not clear that the public and private key belong to different people, what about `their_public_key` and `my_private_key` or something to that effect?